options.go

```diff
@@ -178,6 +187,16 @@ func WithHTTPClient(client *http.Client) Option {
 func WithAdminAPIEndpoint(url string) Option {
 	return func(d *dialerConfig) {
 		d.sqladminOpts = append(d.sqladminOpts, apiopt.WithEndpoint(url))
+		d.serviceUniverseDomain = strings.SplitN(url, ".", 2)[1]
```

Comment:
not sure if this is complex enough logic, just splitting string based on first "." and grabbing suffix as universe domain

Reply:
We can remove this line like we discussed. You might have to track that this option (or universe) was enabled, though, to error correctly later.